packages/remark-custom-blocks/src/index.js

```diff
@@ -1,7 +1,7 @@
 const spaceSeparated = require('space-separated-tokens')
 
 function escapeRegExp (str) {
-  return str.replace(/[-[]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, '\\$&')
+  return str.replace(new RegExp(`[-[\\]{}()*+?.\\\\^$|/]`, 'g'), '\\$&')
```

Comment:
`()` is an invalid capturing group, then `*+?` are three consecutive quantifiers that aren't quantifying anything, they all need to be escaped. `.` also needs to be escaped. Then `|` also needs to be escaped, otherwise it's a logical OR.
`/` doesn't *need* to be escaped but it wouldn't hurt to do so.

Reply:
To me the correct fix here would be:
```diff
- [-[]\/\{\}\(\)\*\+\?\.\\\^\$\|]
+ [-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]
```

Reply:
I used https://regexr.com to check your regex and I think this website got it wrong. Now I'm not sure what is correct and what isn't. I wrote tests and it seems that your regex works properly.

Reply:
I utilized both https://regexr.com and https://regex101.com. I believe I had better results on the later.
I was having trouble with the ESLint rule for unnecessary escapes when escaping all of the characters, and trouble with the regex itself when only escaping the necessary ones utilizing the regex literal, which is why I moved to the RegExp class.